app/assets/javascripts/components/revisions/revisions_handler.jsx

```diff
@@ -3,41 +3,44 @@ import createReactClass from 'create-react-class';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import RevisionList from './revision_list.jsx';
-import { fetchRevisions, sortRevisions, filterCourseSpecificRevisions } from '../../actions/revisions_actions.js';
-import { getActivityRevisions } from '../../selectors';
+import { fetchRevisions, sortRevisions } from '../../actions/revisions_actions.js';
 
 const RevisionHandler = createReactClass({
   displayName: 'RevisionHandler',
 
   propTypes: {
     course_id: PropTypes.string,
     course: PropTypes.object,
+    courseScopedLimit: PropTypes.number,
+    courseScopedLimitReached: PropTypes.bool,
+    courseSpecificRevisions: PropTypes.array,
     fetchRevisions: PropTypes.func,
-    limitReached: PropTypes.bool,
     limit: PropTypes.number,
+    limitReached: PropTypes.bool,
     revisions: PropTypes.array,
-    courseSpecificRevisionIds: PropTypes.array,
     wikidataLabels: PropTypes.object,
     loadingRevisions: PropTypes.bool
   },
 
+  getInitialState() {
+    return {
+      isCourseScoped: false
+    };
+  },
+
   componentDidMount() {
     if (this.props.loadingRevisions) {
       this.props.fetchRevisions(this.props.course_id, this.props.limit);
+      this.props.fetchRevisions(this.props.course_id, this.props.courseScopedLimit, true);
```

Comment:
This should only be fetched upon request, when a user switches to course-scoped; otherwise, we're doing a lot of extra queries that in most cases will not be used.

Reply:
Done! `componentDidUpdate()` was making the most sense for me to do this, so I implemented it there

Reply:
Hmm... not sure that's the best place for it. Why not have it be part of action from clicking the button?

Reply:
Done!